### PR TITLE
[CuTeDSL] Fix SiTU-GLU Hadamard API contracts

### PR DESCRIPTION
**Pull request by harryzhou2000**
## Before submitting

- [x] I agree to license this contribution under the terms of [LICENSE.txt](https://github.com/NVIDIA/cudnn-frontend/blob/develop/LICENSE.txt).
- [x] I ran `pre-commit run` and committed any formatting changes.
- [ ] I added GitHub labels: one `cat-*`, one or more `mod-*`, and one `orig-*` (see [label list](https://github.com/NVIDIA/cudnn-frontend/labels)). Fork authors cannot apply upstream labels; requested labels are `cat-bugfix`, `mod-cutedsl`, and `orig-nv-eng`.

## Affected area

FE OSS kernels or CuTeDSL

## Summary

Follow up on [PR 645](https://github.com/NVIDIA/cudnn-frontend/pull/645) with three API-contract corrections for the NVFP4 grouped GEMM + SiTU-GLU + Hadamard path:

- make omitted `GroupedGemmGluHadamardSm100.execute()` beta arguments inherit the values stored in the compiled API object;
- validate positive, finite SiTU beta values before the high-level wrapper's empty-input fast return;
- include the routing-probability factor in the CuTe kernel's documented SiTU-GLU equation.

The tests add a numerical non-default-beta compile/execute case and empty-input validation cases for zero and NaN beta values.

## Why

`situ_beta1` specializes the compiled kernel, while `situ_beta2` is a runtime scalar. The merged implementation gave `execute()` hard-coded `(4, 25)` defaults even when the API object had been constructed and compiled for another valid pair such as `(2, 8)`. Omitting execute-time values could therefore reject a valid compiled object or launch with the wrong runtime `beta2`.

Separately, the high-level wrapper returned zero-token outputs before constructing the API object, bypassing the documented positive-finite beta validation. The kernel docstring also omitted `prob` even though the epilogue multiplies by it.

## Related issues

Follow-up to [NVIDIA/cudnn-frontend#645](https://github.com/NVIDIA/cudnn-frontend/pull/645).

## API and compatibility impact

- No kernel-math, layout, quantization, or performance-path change.
- Existing callers that explicitly pass beta values are unchanged.
- Callers that omit execute-time beta values now consistently reuse the values with which the API object was constructed.
- Invalid SiTU beta values are rejected consistently for empty and non-empty high-level wrapper inputs.
- Tested with cuDNN Frontend 1.27.0, cuDNN backend 9.23.1, and PyTorch 2.12 development builds on NVIDIA B200.

## Testing

- `python3 -m py_compile` on all three changed Python files: passed.
- `pre-commit run` on the staged files: passed.
- CuTe DSL 4.5.2, B200:
  - full L0 grouped GLU/dGLU/GLU-Hadamard matrix: **818 passed, 692 skipped**;
  - focused GLU-Hadamard file: **22 passed**.
- CuTe DSL 4.6.2, B200:
  - full L0 grouped GLU/dGLU/GLU-Hadamard matrix: **818 passed, 692 skipped**;
  - focused GLU-Hadamard file: **22 passed**.

## Summary by CodeRabbit

* **New Features**
  * Added optional SiT-GLU beta parameters during execution, with automatic fallback to configured values.
  * Added validation requiring beta values to be finite and positive.

* **Documentation**
  * Clarified the SiTU-GLU formula to include the probability term.

* **Bug Fixes**
  * Invalid beta values are now detected consistently, including when processing empty inputs.

### CODE REVIEW

**Comment by harryzhou2000** [comment]
@Anerudhan PR 645 merged while these final review fixes were validating, so I consolidated the three API-contract corrections into this focused follow-up. Could you please apply `cat-bugfix`, `mod-cutedsl`, and `orig-nv-eng`, and launch the cuDNN CI bot `python_tests` target on this head? The full local grouped GLU/dGLU/GLU-Hadamard L0 matrix passed on B200 with both CuTe DSL 4.5.2 and 4.6.2 (818 passed, 692 skipped on each stack).

File: python/cudnn/gemm/cutedsl/grouped/glu_hadamard/api.py [hunk 1]
```diff
@@ -573,15 +573,19 @@ def execute(
...
     ) -> None:
         import torch
 
         self._ensure_support_checked()
         if self._compiled_kernel is None:
             raise RuntimeError("Kernel has not been compiled")
+        if situ_beta1 is None:
+            situ_beta1 = self.situ_beta1
+        if situ_beta2 is None:
+            situ_beta2 = self.situ_beta2
```
[the remark this answers is not in the document]
Comment (harryzhou2000) [reply python/cudnn/gemm/cutedsl/grouped/glu_hadamard/api.py:588]: Fixed in `f422e22f`: `execute()` now validates the resolved SiTU-GLU beta values before the empty-input return, and direct API regression cases cover `situ_beta1=0.0` and `situ_beta2=NaN`. The complete GLU-Hadamard test file passes on B200 with cuDNN frontend 1.27.0 and CuTe DSL 4.6.2 (`25 passed`).

File: test/python/fe_api/grouped_gemm/test_grouped_gemm_glu_hadamard.py [hunk 2]
```diff
@@ -235,9 +250,13 @@ def _run_compile_execute(request, *, ab_dtype, sf_dtype, sf_vec_size, act_func="
...
         post_rht_amax_tensor=outputs["post_rht_amax_tensor"],
         bias_tensor=inputs["bias_tensor"],
+        situ_beta1=execute_situ_beta1,
+        situ_beta2=execute_situ_beta2,
     )
 
-    _check_reference(inputs, outputs, cfg, act_func=act_func)
+    if execute_empty_input:
+        return
+    _check_reference(inputs, outputs, cfg, act_func=act_func, situ_beta1=situ_beta1, situ_beta2=situ_beta2)
```
[the remark this answers is not in the document]
Comment (harryzhou2000) [reply test/python/fe_api/grouped_gemm/test_grouped_gemm_glu_hadamard.py]: Fixed in `45577efa`: the helper now uses execution-time beta overrides when building the numerical reference, and a non-default runtime `situ_beta2=8.0` regression test exercises that contract. The complete GLU-Hadamard file passes on B200 with cuDNN frontend 1.27.0 and CuTe DSL 4.6.2 (`25 passed`).
